### PR TITLE
Refactors Operator Packages

### PR DESCRIPTION
**Pull request by danehans**
Refactors operator packages to support introducing additional controllers, i.e. `gatewaycontroller.New()`.

- `Dockerfile`: Updates `COPY` to support refactored packaging.
- `cmd/contour-operator.go`: CLI flags and default flag values are now obtained from package `config`.
- `controller/contour/controller_test.go` and `controller/contour/finalizer_test.go`: Moved tests to operator package test suite.
- `internal/operator/config/config.go`: New package to manage configuration parameters of the operator.
- `internal/operator/*`: Replaces package `manager`.
- `internal/operator/suite_test.go`: Tests that were moved from the `controller/contour` package. Also updates the tests to create and start the operator from the `operator` package.

Other changes are related to handing type updates or renaming/refactoring of packages.

In support of: https://github.com/projectcontour/contour-operator/pull/171
Fixes: https://github.com/projectcontour/contour-operator/issues/114

Signed-off-by: Daneyon Hansen <daneyonhansen@gmail.com>

### CODE REVIEW

File: internal/operator/config/config.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+// Copyright Project Contour Authors
```
[the remark this answers is not in the document]
Comment (danehans) [reply internal/operator/config/config.go:1]: @skriss `operator` will contain multiple controllers, each with potentially their own config. I like `internal/operator/config/` to identify the operator-level config as apposed to a more abstract "internal" config.

File: pkg/equality/equality_test.go [hunk 2]
```diff
@@ -17,8 +17,8 @@ import (
 	"testing"
```
[the remark this answers is not in the document]
Comment (danehans) [reply pkg/equality/equality_test.go]: @skriss commit `212e800` moves the code from pkg to internal.

**Comment by danehans** [comment]
@skriss thanks for the review. Commit `212e800` updates the PR based on your comments.

**Review by skriss — APPROVED** [review]
no other comments from me, LGTM

File: internal/equality/equality_test.go [hunk 3]
```diff
@@ -17,8 +17,8 @@ import (
 	"testing"
 
 	operatorv1alpha1 "github.com/projectcontour/contour-operator/api/v1alpha1"
-	"github.com/projectcontour/contour-operator/controller/contour"
-	utilequality "github.com/projectcontour/contour-operator/util/equality"
```
Comment (youngnick) [inline internal/equality/equality_test.go:21]: 👏 for removing a `util` package!

**Review by youngnick — APPROVED** [review]
LGTM, I'll move to merging.

**Pull request merged**